packages/runtime/src/helpers/edge.ts

```diff
@@ -90,17 +97,20 @@ const getMiddlewareBundle = async ({
   netlifyConfig: NetlifyConfig
 }): Promise<string> => {
   const { publish } = netlifyConfig.build
-  const chunks: Array<string> = [bootstrap]
+  const chunks: Array<string> = [preamble]
   for (const file of edgeFunctionDefinition.files) {
     const filePath = join(publish, file)
-    const data = await fs.readFile(filePath, 'utf8')
+
+    let data = await fs.readFile(filePath, 'utf8')
+    // Next defines an immutable global variable, which is fine unless you have more than one in the bundle
+    // This adds a check to see if the global is already defined
+    data = IMPORT_UNSUPPORTED.reduce(
+      (acc, val) => acc.replace(val, `('__import_unsupported' in globalThis)||${val}`),
+      data,
+    )
     chunks.push('{', data, '}')
   }
 
-  const middleware = await fs.readFile(join(publish, `server`, `${edgeFunctionDefinition.name}.js`), 'utf8')
```

Comment:
This isn't needed anymore, because the middleware file itself is included in `edgeFunctionDefinition.files`